### PR TITLE
detect-uricontent: convert tests to suricata verify

### PR DESCRIPTION
**Pull request by TheKharleeci**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/4911

Describe changes:
- convert tests to Suricata-verify

suricata-verify-pr: 627

### CODE REVIEW

**Comment by victorjulien** [comment]
I think the git message should not say "convert", as it suggests it is converted to something else in the suricata code. I would say something like:

```
detect/uricontent: remove unittests

These tests are reimplemented as Suricata-Verify tests

Ticket ...
```

File: src/detect-uricontent.c [hunk 1]
```diff
@@ -501,1221 +226,6 @@ static int DetectUriSigTest01(void)
...
-        printf("umatch %p or pmatch %p: ", s->sm_lists[g_http_uri_buffer_id], s->sm_lists[DETECT_SM_LIST_PMATCH]);
-        goto end;
-    }
-
-    if (    ((DetectContentData*) s->sm_lists[DETECT_SM_LIST_PMATCH]->ctx)->depth != 15 ||
-            ((DetectContentData*) s->sm_lists[DETECT_SM_LIST_PMATCH]->ctx)->offset != 5 ||
-            ((DetectContentData*) s->sm_lists_tail[g_http_uri_buffer_id]->ctx)->distance != 30 ||
-            ((DetectContentData*) s->sm_lists_tail[g_http_uri_buffer_id]->ctx)->within != 60 ||
-            ((DetectContentData*) s->sm_lists_tail[DETECT_SM_LIST_PMATCH]->ctx)->distance != 45 ||
-            ((DetectContentData*) s->sm_lists_tail[DETECT_SM_LIST_PMATCH]->ctx)->within != 70 ||
```
Comment (inashivb) [inline src/detect-uricontent.c:942]: @victorjulien is it ok to remove these tests? The signatures are used as a part of tests in s-v so it means they were parsed properly but here we're asserting every param in the signature. Also, it does not involve multi layer integrations testing.

Comment (victorjulien) [reply src/detect-uricontent.c:942]: Right, this shouldn't be removed, but should be converted to use FAIL_IF/PASS logic (in a separate commit)

File: src/detect-uricontent.c [hunk 2]
```diff
@@ -501,1221 +226,6 @@ static int DetectUriSigTest01(void)
...
-    UTHFreeFlow(f);
-    UTHFreePackets(&p, 1);
-    StreamTcpFreeConfig(true);
-    PASS;
-}
-
-/**
- * \test Test content for dce sig.
- */
-static int DetectUriSigTest08(void)
```
Comment (victorjulien) [inline src/detect-uricontent.c:1218]: same for these rule parsing tests

File: src/detect-uricontent.c (hunk 1 above)
Comment (inashivb) [reply src/detect-uricontent.c:942]: @TheKharleeci let's keep these signature parsing tests here intact (the ones that are *only* parsing signatures) even if we have used this signatures in the suricata-verify tests. Thank you!

Comment (TheKharleeci) [reply src/detect-uricontent.c:942]: Ok. Thank you.

**Comment by TheKharleeci** [comment]
Followed by: #6804

**Pull request closed without merging**